webservices/legal_docs/advisory_opinions.py

```diff
@@ -14,24 +14,26 @@
 
 ALL_AOS = """
     SELECT
-        ao_id,
-        ao_no,
-        name,
-        summary,
-        req_date,
-        issue_date,
-        CASE WHEN finished IS NULL THEN TRUE ELSE FALSE END AS is_pending
-    FROM aouser.aos_with_parsed_numbers ao
-    LEFT JOIN (SELECT DISTINCT ao_id AS finished
-               FROM aouser.document
-               WHERE category IN ('Final Opinion', 'Withdrawal of Request')) AS finished
-        ON ao.ao_id = finished.finished
+        ao_parsed.ao_id,
+        ao_parsed.ao_no,
+        ao_parsed.name,
+        ao_parsed.summary,
+        ao_parsed.req_date,
+        ao_parsed.issue_date,
+        CASE ao.stage WHEN 0 THEN TRUE ELSE FALSE END AS is_pending,
```

Comment:
Would it be better to just retrieve `ao.stage` in the SQL query and do the `ao_status` and `ao_is_pending` evaluations in `python`?

Reply:
IMO, the logic would be simpler to understand in Python, and we could also keep this evaluation and the function `ao_status_to_stage` in the same Python file, so that they could be edited together (if there were future changes to be made, or future special cases to be handled like 2009-05).

Reply:
@vrajmohan works for me! Will do.